xcode/Subconscious/Shared/Components/Common/Profile/UserProfileView.swift

```diff
@@ -54,21 +54,19 @@ struct RecentTabView: View {
 struct FollowTabView: View {
     var state: UserProfileDetailModel
     var send: (UserProfileDetailAction) -> Void
-    var onNavigateToUser: (UserProfile) -> Void
+    var onNavigateToUser: (Slashlink) -> Void
     var onProfileAction: (UserProfile, UserProfileAction) -> Void
     
     var body: some View {
         ForEach(state.following) { follow in
             StoryUserView(
                 story: follow,
-                action: { _ in onNavigateToUser(follow.user) },
+                onNavigate: {
+                    onNavigateToUser(follow.user.address)
+                },
                 profileAction: onProfileAction,
                 onRefreshUser: {
-                    guard let petname = follow.user.address.toPetname() else {
-                        return
-                    }
-                    
-                    send(.requestWaitForFollowedUserResolution(petname))
+                    send(.requestWaitForFollowedUserResolution(follow.entry.petname))
```

Comment:
Nice. It's a good smell to move guard logic out of event handlers and into update functions.